app/api/users.py

```diff
@@ -108,9 +108,12 @@ class UserDetail(ResourceDetail):
     """
 
     def before_get(self, args, kwargs):
-        if current_user.is_admin or current_user.is_super_admin or current_user:
```

Comment:
I'd prefer this syntax, what's problem in this?

Reply:
1- current_user is only fetched if the code encounters a jwt_required, excluding event_id from the decorator makes current_user null
2- If the third option allows every current_user why are we even checking is_admin or is_super_admin

Reply:
1. Other endpoints have worked around this by conditionally loading current_user using a decorator. Check those out
2. `or current_user` This section is wrong

Reply:
1- In this section current_user is something differentiating between whether the decorator is_user_itself, considered or not:

- importing it conditionally on the whole if-else check will fail the system to differentiate and will provide complete user detail for all

- importing it on 2nd elif will help with current_user.is_admin or curent_user.is_super_admin check but we will need has_access to check is_user_itself (equivalent to the code I'll update considering suggestion in the next hound comments)

2- Yes it was wrong but current_user check is indeed important for the request coming from decorators to take some form of schema

Reply:
It's done in other places without requiring the Authorization check. Take a look